slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/SlashingProtectionFactory.java

```diff
@@ -22,16 +23,38 @@
 
 public class SlashingProtectionFactory {
 
+  public static final int EXPECTED_DATABASE_VERSION = 6;
+  private static final String errorMsg =
+      String.format(
+          "Database does not have expected version (%s), please run migrations and try again.",
+          EXPECTED_DATABASE_VERSION);
+
   public static SlashingProtection createSlashingProtection(
       final String slashingProtectionDbUrl,
       final String slashingProtectionDbUser,
       final String slashingProtectionDbPassword) {
     final Jdbi jdbi =
         DbConnection.createConnection(
             slashingProtectionDbUrl, slashingProtectionDbUser, slashingProtectionDbPassword);
+
+    verifyVersion(jdbi);
+
     return createSlashingProtection(jdbi);
   }
 
+  private static void verifyVersion(final Jdbi jdbi) {
+    final DatabaseVersionDao databaseVersionDao = new DatabaseVersionDao();
+
+    try {
+      final Integer version = jdbi.withHandle(databaseVersionDao::findDatabaseVersion);
```

Comment:
nit: int instead of Integer, I know will box but then we can use == instead of compareTo

Reply:
prefer not to box, and use == 👍